src/main/kotlin/app/revanced/patcher/extensions/InstructionExtensions.kt

```diff
@@ -11,7 +12,28 @@ import org.jf.dexlib2.builder.MutableMethodImplementation
 import org.jf.dexlib2.builder.instruction.*
 import org.jf.dexlib2.iface.instruction.Instruction
 
+
+class InstructionInserter(val method: MutableMethod, startIndex: Int) {
+    var index = startIndex
+
+    fun addInstructions(smaliInstructions: String) {
+        addInstructions(smaliInstructions.toInstructions(method))
+    }
+
+    fun addInstructions(instructions: List<BuilderInstruction>) {
+        method.implementation!!.addInstructions(index, instructions)
+        index += instructions.size
+    }
+}
+
 object InstructionExtensions {
+    fun MutableMethod.insertInstructions(startIndex: Int, fn: InstructionInserter.() -> Unit): Int {
```

Comment:
Please add comments following the style of the others and move the functions to the correct place.